### PR TITLE
Convert to module, minor cleanup

### PR DESCRIPTION
**Pull request by ravron**
Hi, thanks for the tool! I've done some cleanup you may be interested in. Specifically:

* Add `go.mod` to convert this to a module, making it much easier to run and test
* Somewhat improve the error output, specifically by using the standard `package.Position.String()`, and by printing the full line of code where the error occurs, rather than starting at the byte offset
* Slight test cleanup

Also, I thought I'd mention that it looks like @rickfillion's prior commits were made under rick@centrix.ca rather than an email on the 1Password domain.

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -0,0 +1,27 @@
+github.com/yuin/goldmark v1.3.5/go.mod h1:mwnBkeHKe2W/ZEtQ+71ViKU8L12m81fl3OWwC1Zlc8k=
```
Comment (rickfillion) [inline go.sum:1]: thanks for the go mod support, but it seems odd that goldmark is included here. can you double check that this go.sum file is appropriate given the few dependencies this should actually have

**Review by rickfillion — CHANGES_REQUESTED** [review]
thanks for the improvements! they look great outside of the go.sum file.

regarding my commits being under the wrong email address... good eye! I'll go update my local git config for this repo.

File: go.sum (hunk 1 above)
Comment (ravron) [reply go.sum:1]: Good catch! I believe this is correct, though. While this tool has only one direct dependency listed in `go.mod`, `golang.org/x/tools@v0.1.4`, that dependency [has a few others of its own](https://cs.opensource.google/go/x/tools/+/refs/tags/v0.1.4:go.mod;l=6), including `github.com/yuin/goldmark@v1.3.5`. I agree it's a little odd that `golang.org/x/tools` depends on goldmark, but it does. 

Here's the full output of `go mod graph` (each line of the output represents an edge in the module dependency graph):

```
github.com/1Password/go-directequality-checker golang.org/x/tools@v0.1.4
golang.org/x/tools@v0.1.4 github.com/yuin/goldmark@v1.3.5
golang.org/x/tools@v0.1.4 golang.org/x/mod@v0.4.2
golang.org/x/tools@v0.1.4 golang.org/x/net@v0.0.0-20210405180319-a5a99cb37ef4
golang.org/x/tools@v0.1.4 golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c
golang.org/x/tools@v0.1.4 golang.org/x/sys@v0.0.0-20210510120138-977fb7262007
golang.org/x/tools@v0.1.4 golang.org/x/xerrors@v0.0.0-20200804184101-5ec99f83aff1
golang.org/x/mod@v0.4.2 golang.org/x/crypto@v0.0.0-20191011191535-87dc89f01550
golang.org/x/mod@v0.4.2 golang.org/x/tools@v0.0.0-20191119224855-298f0cb1881e
golang.org/x/mod@v0.4.2 golang.org/x/xerrors@v0.0.0-20191011141410-1b5146add898
golang.org/x/net@v0.0.0-20210405180319-a5a99cb37ef4 golang.org/x/sys@v0.0.0-20210330210617-4fbd30eecc44
golang.org/x/net@v0.0.0-20210405180319-a5a99cb37ef4 golang.org/x/term@v0.0.0-20201126162022-7de9c90e9dd1
golang.org/x/net@v0.0.0-20210405180319-a5a99cb37ef4 golang.org/x/text@v0.3.3
golang.org/x/crypto@v0.0.0-20191011191535-87dc89f01550 golang.org/x/net@v0.0.0-20190404232315-eb5bcb51f2a3
golang.org/x/crypto@v0.0.0-20191011191535-87dc89f01550 golang.org/x/sys@v0.0.0-20190412213103-97732733099d
golang.org/x/tools@v0.0.0-20191119224855-298f0cb1881e golang.org/x/net@v0.0.0-20190620200207-3b0461eec859
golang.org/x/tools@v0.0.0-20191119224855-298f0cb1881e golang.org/x/sync@v0.0.0-20190423024810-112230192c58
golang.org/x/tools@v0.0.0-20191119224855-298f0cb1881e golang.org/x/xerrors@v0.0.0-20190717185122-a985d3407aa7
golang.org/x/term@v0.0.0-20201126162022-7de9c90e9dd1 golang.org/x/sys@v0.0.0-20201119102817-f84b799fce68
golang.org/x/text@v0.3.3 golang.org/x/tools@v0.0.0-20180917221912-90fa682c2a6e
golang.org/x/net@v0.0.0-20190404232315-eb5bcb51f2a3 golang.org/x/crypto@v0.0.0-20190308221718-c2843e01d9a2
golang.org/x/net@v0.0.0-20190404232315-eb5bcb51f2a3 golang.org/x/text@v0.3.0
golang.org/x/net@v0.0.0-20190620200207-3b0461eec859 golang.org/x/crypto@v0.0.0-20190308221718-c2843e01d9a2
golang.org/x/net@v0.0.0-20190620200207-3b0461eec859 golang.org/x/sys@v0.0.0-20190215142949-d0b11bdaac8a
golang.org/x/net@v0.0.0-20190620200207-3b0461eec859 golang.org/x/text@v0.3.0
golang.org/x/crypto@v0.0.0-20190308221718-c2843e01d9a2 golang.org/x/sys@v0.0.0-20190215142949-d0b11bdaac8a
```

**Comment by ravron** [comment]
Hey, just keen to clean up old PRs. If you're still interested in this work, kindly re-review it in light of my [comment](https://github.com/1Password/go-directequality-checker/pull/1#discussion_r659078125). If not, no problem — I'll close it in the next few weeks.

**Pull request closed without merging**